ADALiOS/ADALiOS/ADLogger.m

```diff
@@ -126,6 +126,8 @@ +(void) log: (ADAL_LOG_LEVEL)logLevel
                 sLogCallback(logLevel, [NSString stringWithFormat:@"ADALiOS [%@ - %@] %@", [dateFormatter stringFromDate:[NSDate date]], [[ADLogger getCorrelationId] UUIDString], message], additionalInformation, errorCode);
             }
         }
+        SAFE_ARC_RELEASE(dateFormatter);
```

Comment:
Sure. I could see it causing compiler errors if the compiler is being overly aggressive. It’s not high pri, just an idea. ☺

From: Kanishk Panwar [mailto:notifications@github.com]
Sent: Thursday, February 19, 2015 3:41 PM
To: AzureAD/azure-activedirectory-library-for-objc
Cc: Ryan Pangrle
Subject: Re: [azure-activedirectory-library-for-objc] memory leak fixes (#271)


In ADALiOS/ADALiOS/ADLogger.m<https://github.com/AzureAD/azure-activedirectory-library-for-objc/pull/271#discussion_r25037814>:

> @@ -126,6 +126,8 @@ +(void) log: (ADAL_LOG_LEVEL)logLevel

>                  sLogCallback(logLevel, [NSString stringWithFormat:@"ADALiOS [%@ - %@] %@", [dateFormatter stringFromDate:[NSDate date]], [[ADLogger getCorrelationId] UUIDString], message], additionalInformation, errorCode);

>              }

>          }

> +        SAFE_ARC_RELEASE(dateFormatter);

will it be okay if I put nil after the release here? I think it will require a major sweep across the code else we can get double nil assignment errors.

—
Reply to this email directly or view it on GitHub<https://github.com/AzureAD/azure-activedirectory-library-for-objc/pull/271/files#r25037814>.